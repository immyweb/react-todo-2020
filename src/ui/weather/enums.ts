export enum Locations {
  WICK = 'WICK',
  ABERDEEN = 'ABERDEEN',
  EDINBURGH = 'EDINBURGH',
  BELFAST = 'BELFAST',
  MANCHESTER = 'MANCHESTER',
  CARDIFF = 'CARDIFF',
  PLYMOUTH = 'PLYMOUTH',
  LONDON = 'LONDON',
}

export enum Days {
  MON = 'MON',
  TUE = 'TUE',
  WED = 'WED',
  THU = 'THU',
  FRI = 'FRI',
  SAT = 'SAT',
  SUN = 'SUN',
}

export enum Direction {
  LEFT = 'LEFT',
  RIGHT = 'RIGHT',
  TOP = 'TOP',
  BOTTOM = 'BOTTOM',
}

export enum WeatherIconUrls {
  CLEAR_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/clear-night@2x1.png',
  SUNNY_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png',
  PARTLY_CLOUDY_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Party-Cloudy.png',
  PARTLY_CLOUDY_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Party-Cloudy.png',
  MIST = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/foggy@2x1.png',
  FOG = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/foggy@2x1.png',
  CLOUDY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Cloudy.png',
  OVERCAST = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Cloudy.png',
  LIGHT_RAIN_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  LIGHT_RAIN_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  DRIZZLE = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  LIGHT_RAIN = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  HEAVY_RAIN_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  HEAVY_RAIN_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  HEAVY_RAIN = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Rain2.png',
  SLEET_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  SLEET_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  SLEET = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  HAIL_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Hail.png',
  HAIL_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Hail.png',
  HAIL = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Hail.png',
  LIGHT_SNOW_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  LIGHT_SNOW_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  LIGHT_SNOW = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  HEAVY_SNOW_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  HEAVY_SNOW_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  HEAVY_SNOW = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Snow.png',
  THUNDER_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Storm.png',
  THUNDER_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Storm.png',
  THUNDER = 'https://www.thesun.co.uk/wp-content/uploads/2018/11/Storm.png',
}

export enum WeatherAtmosphereUrls {
  CLEAR_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_clearnight@2x.jpg',
  SUNNY_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_sunny@2x.jpg',
  PARTLY_CLOUDY_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_partlycloudynight@2x.jpg',
  PARTLY_CLOUDY_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_partlycloudynight@2x.jpg',
  MIST = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_foggy@2x.jpg',
  FOG = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_foggy@2x.jpg',
  CLOUDY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_cloudy@2x.jpg',
  OVERCAST = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_cloudy@2x.jpg',
  LIGHT_RAIN_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  LIGHT_RAIN_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  DRIZZLE = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  LIGHT_RAIN = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  HEAVY_RAIN_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  HEAVY_RAIN_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  HEAVY_RAIN = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  SLEET_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  SLEET_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  SLEET = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  HAIL_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  HAIL_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  HAIL = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_raining@2x.jpg',
  LIGHT_SNOW_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  LIGHT_SNOW_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  LIGHT_SNOW = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  HEAVY_SNOW_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  HEAVY_SNOW_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  HEAVY_SNOW = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_snowing@2x.jpg',
  THUNDER_SHOWER_NIGHT = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_stormy@2x.jpg',
  THUNDER_SHOWER_DAY = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_stormy@2x.jpg',
  THUNDER = 'https://www.thesun.co.uk/wp-content/uploads/2018/08/BG_stormy@2x.jpg',
}
