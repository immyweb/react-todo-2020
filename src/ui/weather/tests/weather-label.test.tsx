import * as React from 'react';
import { shallow } from 'enzyme';
import { WeatherLabel } from '../weather-label';
import { WeatherIconUrls, Direction } from '../enums';
import { colours } from '../../../styles/colours';

const defaultWeatherLabelProps = {
  dayMax: '10',
  nightMin: '5',
  temperatureColour: colours.$white,
  temperatureBackgroundColour: colours.$mainSecondary,
  labelColour: colours.$white,
  labelBackgroundColour: colours.$offBlack,
  direction: Direction.LEFT,
  arrowDirection: Direction.LEFT,
  icon: WeatherIconUrls.SUNNY_DAY,
  mobilePosition: {
    top: '5%',
    left: '15%',
  },
  desktopPosition: {
    top: '5%',
    left: '15%',
  },
};

describe('<WeatherLabel />', () => {
  it('renders correctly with default values', () => {
    const wrapper = shallow(<WeatherLabel {...defaultWeatherLabelProps}>SWANSEA</WeatherLabel>);
    expect(wrapper).toMatchSnapshot();
  });
  it('renders in reverse when direction is RIGHT', () => {
    const wrapper = shallow(
      <WeatherLabel {...defaultWeatherLabelProps} direction={Direction.RIGHT}>
        SWANSEA
      </WeatherLabel>,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
