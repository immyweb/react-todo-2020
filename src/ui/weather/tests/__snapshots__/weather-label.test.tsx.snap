// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WeatherLabel /> renders correctly with default values 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    arrowDirection="LEFT"
    dayMax="10"
    desktopPosition={
      Object {
        "left": "15%",
        "top": "5%",
      }
    }
    direction="LEFT"
    icon="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
    labelBackgroundColour="#222526"
    labelColour="#ffffff"
    mobilePosition={
      Object {
        "left": "15%",
        "top": "5%",
      }
    }
    nightMin="5"
    temperatureBackgroundColour="#a5171c"
    temperatureColour="#ffffff"
  >
    SWANSEA
  </Unknown>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <glamorous(div)
          direction="LEFT"
        >
          <div
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <glamorous(div)
              className="WeatherLabelIcon"
              iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
            />
            <glamorous(div)
              backgroundColour="#222526"
              className="WeatherLabelArrow"
              direction="LEFT"
            />
          </div>
          <glamorous(div)
            className="WeatherLabelName"
            labelBackgroundColour="#222526"
            labelColour="#ffffff"
          >
            SWANSEA
          </glamorous(div)>
        </glamorous(div)>,
        <glamorous(div)>
          <glamorous(div)
            className="WeatherLabelTemperature"
            temperatureBackgroundColour="#a5171c"
            temperatureColour="#ffffff"
          >
            10°
          </glamorous(div)>
          <glamorous(div)
            className="WeatherLabelTemperature"
            temperatureBackgroundColour="#2d5c8c"
            temperatureColour="#ffffff"
          >
            5°
          </glamorous(div)>
        </glamorous(div)>,
      ],
      "className": "WeatherLabelContainer",
      "dLeft": "15%",
      "dTop": "5%",
      "direction": "LEFT",
      "mLeft": "15%",
      "mTop": "5%",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <glamorous(div)
                className="WeatherLabelIcon"
                iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
              />
              <glamorous(div)
                backgroundColour="#222526"
                className="WeatherLabelArrow"
                direction="LEFT"
              />
            </div>,
            false,
            false,
            false,
            <glamorous(div)
              className="WeatherLabelName"
              labelBackgroundColour="#222526"
              labelColour="#ffffff"
            >
              SWANSEA
            </glamorous(div)>,
          ],
          "direction": "LEFT",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <glamorous(div)
                  className="WeatherLabelIcon"
                  iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
                />,
                <glamorous(div)
                  backgroundColour="#222526"
                  className="WeatherLabelArrow"
                  direction="LEFT"
                />,
              ],
              "style": Object {
                "display": "flex",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "WeatherLabelIcon",
                  "iconUrl": "https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "backgroundColour": "#222526",
                  "className": "WeatherLabelArrow",
                  "direction": "LEFT",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          false,
          false,
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "SWANSEA",
              "className": "WeatherLabelName",
              "labelBackgroundColour": "#222526",
              "labelColour": "#ffffff",
            },
            "ref": null,
            "rendered": "SWANSEA",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(div)
              className="WeatherLabelTemperature"
              temperatureBackgroundColour="#a5171c"
              temperatureColour="#ffffff"
            >
              10°
            </glamorous(div)>,
            <glamorous(div)
              className="WeatherLabelTemperature"
              temperatureBackgroundColour="#2d5c8c"
              temperatureColour="#ffffff"
            >
              5°
            </glamorous(div)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "10°",
              "className": "WeatherLabelTemperature",
              "temperatureBackgroundColour": "#a5171c",
              "temperatureColour": "#ffffff",
            },
            "ref": null,
            "rendered": "10°",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "5°",
              "className": "WeatherLabelTemperature",
              "temperatureBackgroundColour": "#2d5c8c",
              "temperatureColour": "#ffffff",
            },
            "ref": null,
            "rendered": "5°",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <glamorous(div)
            direction="LEFT"
          >
            <div
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <glamorous(div)
                className="WeatherLabelIcon"
                iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
              />
              <glamorous(div)
                backgroundColour="#222526"
                className="WeatherLabelArrow"
                direction="LEFT"
              />
            </div>
            <glamorous(div)
              className="WeatherLabelName"
              labelBackgroundColour="#222526"
              labelColour="#ffffff"
            >
              SWANSEA
            </glamorous(div)>
          </glamorous(div)>,
          <glamorous(div)>
            <glamorous(div)
              className="WeatherLabelTemperature"
              temperatureBackgroundColour="#a5171c"
              temperatureColour="#ffffff"
            >
              10°
            </glamorous(div)>
            <glamorous(div)
              className="WeatherLabelTemperature"
              temperatureBackgroundColour="#2d5c8c"
              temperatureColour="#ffffff"
            >
              5°
            </glamorous(div)>
          </glamorous(div)>,
        ],
        "className": "WeatherLabelContainer",
        "dLeft": "15%",
        "dTop": "5%",
        "direction": "LEFT",
        "mLeft": "15%",
        "mTop": "5%",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <glamorous(div)
                  className="WeatherLabelIcon"
                  iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
                />
                <glamorous(div)
                  backgroundColour="#222526"
                  className="WeatherLabelArrow"
                  direction="LEFT"
                />
              </div>,
              false,
              false,
              false,
              <glamorous(div)
                className="WeatherLabelName"
                labelBackgroundColour="#222526"
                labelColour="#ffffff"
              >
                SWANSEA
              </glamorous(div)>,
            ],
            "direction": "LEFT",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <glamorous(div)
                    className="WeatherLabelIcon"
                    iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
                  />,
                  <glamorous(div)
                    backgroundColour="#222526"
                    className="WeatherLabelArrow"
                    direction="LEFT"
                  />,
                ],
                "style": Object {
                  "display": "flex",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "WeatherLabelIcon",
                    "iconUrl": "https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "backgroundColour": "#222526",
                    "className": "WeatherLabelArrow",
                    "direction": "LEFT",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            false,
            false,
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "SWANSEA",
                "className": "WeatherLabelName",
                "labelBackgroundColour": "#222526",
                "labelColour": "#ffffff",
              },
              "ref": null,
              "rendered": "SWANSEA",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(div)
                className="WeatherLabelTemperature"
                temperatureBackgroundColour="#a5171c"
                temperatureColour="#ffffff"
              >
                10°
              </glamorous(div)>,
              <glamorous(div)
                className="WeatherLabelTemperature"
                temperatureBackgroundColour="#2d5c8c"
                temperatureColour="#ffffff"
              >
                5°
              </glamorous(div)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "10°",
                "className": "WeatherLabelTemperature",
                "temperatureBackgroundColour": "#a5171c",
                "temperatureColour": "#ffffff",
              },
              "ref": null,
              "rendered": "10°",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "5°",
                "className": "WeatherLabelTemperature",
                "temperatureBackgroundColour": "#2d5c8c",
                "temperatureColour": "#ffffff",
              },
              "ref": null,
              "rendered": "5°",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`<WeatherLabel /> renders in reverse when direction is RIGHT 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    arrowDirection="LEFT"
    dayMax="10"
    desktopPosition={
      Object {
        "left": "15%",
        "top": "5%",
      }
    }
    direction="RIGHT"
    icon="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
    labelBackgroundColour="#222526"
    labelColour="#ffffff"
    mobilePosition={
      Object {
        "left": "15%",
        "top": "5%",
      }
    }
    nightMin="5"
    temperatureBackgroundColour="#a5171c"
    temperatureColour="#ffffff"
  >
    SWANSEA
  </Unknown>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <glamorous(div)
          direction="RIGHT"
        >
          <div
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <glamorous(div)
              className="WeatherLabelIcon"
              iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
            />
            <glamorous(div)
              backgroundColour="#222526"
              className="WeatherLabelArrow"
              direction="LEFT"
            />
          </div>
          <glamorous(div)
            className="WeatherLabelName"
            labelBackgroundColour="#222526"
            labelColour="#ffffff"
          >
            SWANSEA
          </glamorous(div)>
        </glamorous(div)>,
        <glamorous(div)>
          <glamorous(div)
            className="WeatherLabelTemperature"
            temperatureBackgroundColour="#a5171c"
            temperatureColour="#ffffff"
          >
            10°
          </glamorous(div)>
          <glamorous(div)
            className="WeatherLabelTemperature"
            temperatureBackgroundColour="#2d5c8c"
            temperatureColour="#ffffff"
          >
            5°
          </glamorous(div)>
        </glamorous(div)>,
      ],
      "className": "WeatherLabelContainer",
      "dLeft": "15%",
      "dTop": "5%",
      "direction": "RIGHT",
      "mLeft": "15%",
      "mTop": "5%",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <div
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <glamorous(div)
                className="WeatherLabelIcon"
                iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
              />
              <glamorous(div)
                backgroundColour="#222526"
                className="WeatherLabelArrow"
                direction="LEFT"
              />
            </div>,
            false,
            false,
            false,
            <glamorous(div)
              className="WeatherLabelName"
              labelBackgroundColour="#222526"
              labelColour="#ffffff"
            >
              SWANSEA
            </glamorous(div)>,
          ],
          "direction": "RIGHT",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <glamorous(div)
                  className="WeatherLabelIcon"
                  iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
                />,
                <glamorous(div)
                  backgroundColour="#222526"
                  className="WeatherLabelArrow"
                  direction="LEFT"
                />,
              ],
              "style": Object {
                "display": "flex",
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "WeatherLabelIcon",
                  "iconUrl": "https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "backgroundColour": "#222526",
                  "className": "WeatherLabelArrow",
                  "direction": "LEFT",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          false,
          false,
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "SWANSEA",
              "className": "WeatherLabelName",
              "labelBackgroundColour": "#222526",
              "labelColour": "#ffffff",
            },
            "ref": null,
            "rendered": "SWANSEA",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(div)
              className="WeatherLabelTemperature"
              temperatureBackgroundColour="#a5171c"
              temperatureColour="#ffffff"
            >
              10°
            </glamorous(div)>,
            <glamorous(div)
              className="WeatherLabelTemperature"
              temperatureBackgroundColour="#2d5c8c"
              temperatureColour="#ffffff"
            >
              5°
            </glamorous(div)>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "10°",
              "className": "WeatherLabelTemperature",
              "temperatureBackgroundColour": "#a5171c",
              "temperatureColour": "#ffffff",
            },
            "ref": null,
            "rendered": "10°",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "5°",
              "className": "WeatherLabelTemperature",
              "temperatureBackgroundColour": "#2d5c8c",
              "temperatureColour": "#ffffff",
            },
            "ref": null,
            "rendered": "5°",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <glamorous(div)
            direction="RIGHT"
          >
            <div
              style={
                Object {
                  "display": "flex",
                }
              }
            >
              <glamorous(div)
                className="WeatherLabelIcon"
                iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
              />
              <glamorous(div)
                backgroundColour="#222526"
                className="WeatherLabelArrow"
                direction="LEFT"
              />
            </div>
            <glamorous(div)
              className="WeatherLabelName"
              labelBackgroundColour="#222526"
              labelColour="#ffffff"
            >
              SWANSEA
            </glamorous(div)>
          </glamorous(div)>,
          <glamorous(div)>
            <glamorous(div)
              className="WeatherLabelTemperature"
              temperatureBackgroundColour="#a5171c"
              temperatureColour="#ffffff"
            >
              10°
            </glamorous(div)>
            <glamorous(div)
              className="WeatherLabelTemperature"
              temperatureBackgroundColour="#2d5c8c"
              temperatureColour="#ffffff"
            >
              5°
            </glamorous(div)>
          </glamorous(div)>,
        ],
        "className": "WeatherLabelContainer",
        "dLeft": "15%",
        "dTop": "5%",
        "direction": "RIGHT",
        "mLeft": "15%",
        "mTop": "5%",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <glamorous(div)
                  className="WeatherLabelIcon"
                  iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
                />
                <glamorous(div)
                  backgroundColour="#222526"
                  className="WeatherLabelArrow"
                  direction="LEFT"
                />
              </div>,
              false,
              false,
              false,
              <glamorous(div)
                className="WeatherLabelName"
                labelBackgroundColour="#222526"
                labelColour="#ffffff"
              >
                SWANSEA
              </glamorous(div)>,
            ],
            "direction": "RIGHT",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <glamorous(div)
                    className="WeatherLabelIcon"
                    iconUrl="https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png"
                  />,
                  <glamorous(div)
                    backgroundColour="#222526"
                    className="WeatherLabelArrow"
                    direction="LEFT"
                  />,
                ],
                "style": Object {
                  "display": "flex",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "WeatherLabelIcon",
                    "iconUrl": "https://www.thesun.co.uk/wp-content/uploads/2018/11/Sunny.png",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "backgroundColour": "#222526",
                    "className": "WeatherLabelArrow",
                    "direction": "LEFT",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            false,
            false,
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "SWANSEA",
                "className": "WeatherLabelName",
                "labelBackgroundColour": "#222526",
                "labelColour": "#ffffff",
              },
              "ref": null,
              "rendered": "SWANSEA",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(div)
                className="WeatherLabelTemperature"
                temperatureBackgroundColour="#a5171c"
                temperatureColour="#ffffff"
              >
                10°
              </glamorous(div)>,
              <glamorous(div)
                className="WeatherLabelTemperature"
                temperatureBackgroundColour="#2d5c8c"
                temperatureColour="#ffffff"
              >
                5°
              </glamorous(div)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "10°",
                "className": "WeatherLabelTemperature",
                "temperatureBackgroundColour": "#a5171c",
                "temperatureColour": "#ffffff",
              },
              "ref": null,
              "rendered": "10°",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "5°",
                "className": "WeatherLabelTemperature",
                "temperatureBackgroundColour": "#2d5c8c",
                "temperatureColour": "#ffffff",
              },
              "ref": null,
              "rendered": "5°",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
